OSDOCS-3503: RelNotes for Defining Kuryr ports pools & CIDR Ranges (#46521)
openshift/openshift-docs: release_notes/ocp-4-11-release-notes.adoc

For more information, see link:https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy[Service Internal Traffic Policy].

[id="ocp-4-11-OpenShift-Container-Platform-Cluster-Sizes-for-Namespace"]
==== {product-title} Cluster Sizes for Namespaces

> Be aware that CIDR sizes for namespaces are not adjustable post-deployment. Red Hat does not provide direct guidance on determining the size as it requires careful consideration of the number of pods created in the namespace.

Reviewer: +1 Each Kuryr creates of Network and Subnet for each Namespace.

> Be aware that CIDR sizes for namespaces are not adjustable post-deployment. Red Hat does not provide direct guidance on determining the size as it requires careful consideration of the number of pods created in the namespace.

Reviewer 2: You also need to mention both Kuryr and OpenStack in this section - since this issue has a really specific scope, you need to call it out.

> Be aware that CIDR sizes for namespaces are not adjustable post-deployment. Red Hat does not provide direct guidance on determining the size as it requires careful consideration of the number of pods created in the namespace.

Author: @MaysaMacedo I apologize for the delay. I've been having issues with merge conflicts that's now resolved. I have updated the PR for a second review. :)

For more information, see link:https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy[Service Internal Traffic Policy].

> <<<<<<< HEAD

Reviewer 2: Please remove this line.

Author: Removed! :+1: :)

Ensure that you migrate from TP to GA. The upstream version of `ExternalDNS` for an {product-title} 4.11 is `v0.12.0` and for the TP is `v0.10.2`.
For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].
=======
[id="ocp-4-11-cluster-ranges-for-network"]
==== {product-title} Cluster Ranges for Networks
Be aware that CIDR ranges for networks are not adjustable after cluster installation. Red Hat does not provide direct guidance on determining the size as it requires careful consideration of the number of pods created.

> >>>>>>> 386ef90f88 (OSDOCS-3726 Define Maximum Number of Neutron Ports)

Reviewer: leftover from rebase/merge?

Author: Correct, thanks for catching this! :)

Ensure that you migrate from TP to GA. The upstream version of `ExternalDNS` for an {product-title} 4.11 is `v0.12.0` and for the TP is `v0.10.2`.
For more information, see xref:../networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc#nw-installing-external-dns-operator-on-cloud-providers[About the External DNS Operator].

[id="ocp-4-11-cidr-ranges-for-network"]
==== {product-title} CIDR Ranges for Networks

> Be aware that CIDR ranges for networks are not adjustable after cluster installation. Red Hat does not provide direct guidance on determining the size as it requires careful consideration of the number of pods created.

Reviewer 2: ```suggestion
Be aware that CIDR ranges for networks are not adjustable after cluster installation. Red Hat does not provide direct guidance on determining the size because it requires careful consideration of the number of pods created.
```

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * In the Red Hat OpenStack Platform ({rh-openstack}), a potential issue in {rh-openstack} Kuryr may occur when a large variety of Bulk Ports requests are created at the same time. These bulk requests can create conflicts when distributing IP addresses. This issue was resolved in a previous errata; however, Red Hat suggests setting a value to avoid the issue of having a variety of Bulk Ports requests, that is, having multiple pods created simultaneously. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2024690[*BZ2024690*])

Reviewer 2: ```suggestion
* In {rh-openstack-first}, a potential issue in {rh-openstack} Kuryr can occur when a large variety of Bulk Ports requests are created at the same time. These bulk requests can create conflicts when distributing IP addresses. This issue was resolved in a previous errata; however, Red Hat suggests setting a value to avoid the issue of having a variety of Bulk Ports requests, that is, having multiple pods created simultaneously. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2024690[*BZ2024690*])
```

> ! may verb
Use only to indicate permission. Otherwise, use "can" to indicate ability, or use "might" to indicate possibility.

What value? I'm not sure what parameter or value to set.

Are you sure that Bulk Ports needs to be capitalized?

Author: Hi @kalexand-rh, 

After discussing with @MaysaMacedo, it was suggested to me to avoid setting a value and recommend users set their own values. I believe the issue is that users have not been setting their own value.

I have updated the PR to be more clear in suggesting users set their own value, but wanted to ask if you have any recommendations on how I can inform users set a value?

Reviewer 2: @Wiharris, it's more that it's not clear _which parameter_ the user is suppose to set. It sounds like it's something in Kuryr about pods, ports, requests, or some combination of these. If it's a specific known "lever" that we are telling them to "pull," it's useful to point that out.

Also, please do use the `{rh-openstack-first}` attribute. Based on the `common-attributes.adoc` file, it's the best choice for the string you're going for:
> :rh-openstack-first: Red Hat OpenStack Platform (RHOSP)

Author: @kalexand-rh I understand, thank you for catching and sorry about that! I have updated the PR to reflect the specific parameter! :)

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * In the Red Hat OpenStack Platform ({rh-openstack}), a known potential issue in {rh-openstack} Kuryr may occur when attempting to create `300` ports in bulks of `10` ports and `10` subnets. These bulk requests can create conflicts when distributing IP addresses. This issue was resolved in a previous errata; however, Red Hat suggests setting a value lower than `300` to avoid this issue in future releases. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2024690[*BZ2024690*])

Reviewer: It might be better not to mention these numbers because it's very likely that is possible to hit with other values.

Author: @MaysaMacedo That makes complete sense! I have updated the PR to reflect the suggested changes! :)

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * In the {rh-openstack-first}, a potential issue in Kuryr ports pools can occur when a large variety of port creation requests are made at the same time. These bulk port creation requests can create conflicts when distributing IP addresses. This issue was resolved in a previous errata; however, Red Hat suggests setting your own value to avoid the issue of having a variety of bulk port creation requests, that is, having multiple pod ports created simultaneously. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2024690[*BZ2024690*])

Reviewer: > These bulk port creation requests can create conflicts when distributing IP addresses

Perhaps this can be substituted with:
During these bulk ports creation if the IP allocation for one of the IPs fails, the operation will be retried for all ports.

> Red Hat suggests setting your own value to avoid the issue of having a variety of bulk port creation requests

Perhaps the following can be mentioned:
Red Hat suggests setting a small value for the ports pool batch setting to try avoiding a variety of large bulk port creation requests

Author: Thank you for your help and reivew @MaysaMacedo I made the above updated changes! :)

// TODO: This known issue should carry forward to 4.9 and beyond!
* The `oc annotate` command does not work for LDAP group names that contain an equal sign (`=`), because the command uses the equal sign as a delimiter between the annotation name and value. As a workaround, use `oc patch` or `oc edit` to add the annotation. (link:https://bugzilla.redhat.com/show_bug.cgi?id=1917280[*BZ#1917280*])

> * In the {rh-openstack-first}, a potential issue in Kuryr ports pools can occur when a variety of bulk port creation requests are made at the same time. During these bulk requests, if the IP allocation for one of the IPs fails, the operation will be retried for all ports. This issue was resolved in a previous errata; however, Red Hat suggests setting a small value for the ports pool batch settings to avoid a variety of bulk port creation requests. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2024690[*BZ2024690*])

Reviewer 3: ```suggestion
* In the {rh-openstack-first}, a potential issue in Kuryr ports pools can occur when a variety of bulk port creation requests are made at the same time. During these bulk requests, if the IP allocation for one of the IP addresses fails, the operation will be retried for all ports. This issue was resolved in a previous errata; however, Red Hat suggests setting a small value for the ports pool batch settings to avoid a variety of bulk port creation requests. (link:https://bugzilla.redhat.com/show_bug.cgi?id=2024690[*BZ2024690*])
```
